pmlc/target/x86/xsmm_stencil.cc

```diff
@@ -163,6 +163,36 @@ class StencilXSMM : public StencilBase {
       IVLOG(6, debugString(perm.indexes[i]) << ": " << tileSize[i]);
     }
 
+    auto verifyStride = [&](auto op) -> bool {
+      auto si = computeStrideInfo(op.getAffineMap(), op.getMapOperands());
+      if (!si)
+        return false;
+      for (size_t i = 0; i < getTiledIdxCount(); ++i) {
+        for (size_t j = 0; j < si->size(); j++) {
+          if ((*si)[j].strides.count(perm.indexes[i]) == 0) {
+            continue;
+          }
+          if ((*si)[j].strides[perm.indexes[i]] != 1) {
+            IVLOG(3, "Bad stride = " << (*si)[j].strides[perm.indexes[i]]);
+            return false;
+          }
+        }
+      }
+      return true;
+    };
+    for (auto op : perm.ioOps) {
+      bool good = false;
+      if (auto loadOp = mlir::dyn_cast<AffineLoadOp>(op)) {
+        good = verifyStride(loadOp);
+      } else if (auto reduceOp = mlir::dyn_cast<AffineReduceOp>(op)) {
+        good = verifyStride(reduceOp);
+      }
```

Comment:
Do we need to consider `AffineStoreOp`s too?

Reply:
Not for XSMM, since it's a requirement that it have 2 loads + 1 reduce (in the capture() function).